python/paddle/fluid/dygraph/static_runner.py

```diff
@@ -0,0 +1,637 @@
+            optimizer.minimize(avg_loss)
+
+            exe = fluid.Executor(place)
+            exe.run(fluid.default_startup_program())
+
+            loader = fluid.io.DataLoader.from_generator(
+                feed_list=[img, label], capacity=5, iterable=True)
+            loader.set_batch_generator(random_batch_reader(), places=place)
+
+            for data in loader():
+                exe.run(
+                    fluid.default_main_program(),
+                    feed=data, 
+                    fetch_list=[avg_loss])
+
+            # save model by fluid.io.save_inference_model
+            fluid.io.save_inference_model(
+                SAVE_DIRNAME, ["img"], [pred], exe)
+
+
+        # Step 1. train and save inference model in static graph mode
+        place = fluid.CPUPlace()
+        train_and_save_static_model(place)
+
+        # Step 2. load inference model in dygraph and fine-tune
+        with fluid.dygraph.guard(place):
+            fc = fluid.dygraph.static_runner.StaticModelRunner(SAVE_DIRNAME)
+
+            sgd = fluid.optimizer.SGD(learning_rate=0.001,
+                                    parameter_list=fc.parameters())
+
+            train_loader = fluid.io.DataLoader.from_generator(capacity=5)
+            train_loader.set_batch_generator(
+                random_batch_reader(), places=place)
+
+            for data in train_loader():
+                img = data[0]
+                label = data[1]
+                label.stop_gradient = True
+
+                cost = fc(inputs=img)
+
+                loss = fluid.layers.cross_entropy(cost, label)
+                avg_loss = fluid.layers.mean(loss)
+
+                avg_loss.backward()
+                sgd.minimize(avg_loss)
+    """
+
+    def __init__(self, model_dir, model_filename=None, params_filename=None):
+        super(StaticModelRunner, self).__init__()
+
+        # Step 0. key variable definitions
+        self._load_program_desc = None
+        self._program_desc = None
+        # the layer outputs var desc
+        self._output_descs = []
+        # input, output, params name list
+        self._input_names = []
+        self._output_names = []
+        self._param_names = []
+
+        # Step 1. load program desc from disk
+        # the saved model hold feed, fetch & scale op, no need, can be remove
+        self._load_program_desc = self._load_static_model(model_dir,
+                                                          model_filename)
+
+        # Step 2. set all `is_test` attributes to False
+        self._change_is_test_status(False)
+
+        # Step 3. load all parameters
+        self._load_persisitable_dict(model_dir, params_filename)
+
+        # Step 4. generate backwar program desc
+        self._program_desc = self._append_backward_desc()
+
+        # Step 5. recheck parameters stop gradients
+        self._recheck_stop_gradients()
+
+        # For Debug
+        # DescParser.print_program_desc(self._program_desc)
+
+    def train(self):
```

Comment:
Users should call this method explicitly ?

Reply:
I will fix this after https://github.com/PaddlePaddle/Paddle/pull/22463 merged